wp_api_integration_tests/src/lib.rs

```diff
@@ -148,7 +148,7 @@ pub trait AssertResponse {
     fn assert_response(self) -> Self::Item;
 }
 
-impl<T: std::fmt::Debug, E: std::error::Error> AssertResponse for Result<T, E> {
+impl<T: std::fmt::Debug, E: std::fmt::Display + std::fmt::Debug> AssertResponse for Result<T, E> {
```

Comment:
I was a little confused about this change when I first saw it, but didn't raise it because I didn't understand why it was required. I only saw [this comment](https://github.com/Automattic/wordpress-rs/pull/594/files#r2010983929) just now and It think this is the reason for it. Once the error types derive `thiserror::Error`, we can revert this change.